### PR TITLE
Make dd and date plan 9 compatible

### PR DESCRIPTION
**Pull request by rminnich**

### CODE REVIEW

**Comment by rminnich** [comment]
oops I misunderstood the flag. Will fix tomorrow.

File: cmds/core/date/date_linux.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Copyright 2015-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build !plan9
```
Comment (hugelgupf) [inline cmds/core/date/date_linux.go:5]: remove this, it's implied from the filename

File: cmds/core/dd/dd.go [hunk 2]
```diff
@@ -63,21 +62,24 @@ var (
 	bytesWritten int64 // access atomically, must be global for correct alignedness
 )
 
-const (
-	convNoTrunc = 1 << iota
-	oFlagSync
-	oFlagDSync
-)
+type setclear struct {
```
Comment (hugelgupf) [inline cmds/core/dd/dd.go:65]: `setclear` is not a descriptive name; in fact I thought it meant it'd always clear. flagSetter?

File: cmds/core/dd/dd.go [hunk 3]
```diff
@@ -63,21 +62,24 @@ var (
...
 }
 
-var flagMap = map[string]int{
-	"sync":  oFlagSync,
-	"dsync": oFlagDSync,
+var flagMap = map[string]setclear{
+	"sync": {set: os.O_SYNC},
 }
 
+var permflags = os.O_TRUNC | os.O_SYNC
```
Comment (hugelgupf) [inline cmds/core/dd/dd.go:81]: permflags sounds like permanentFlags. allowedFlags? permittedFlags if you spell it out?

File: cmds/core/date/date_linux.go (hunk 1 above)
Comment (rminnich) [reply cmds/core/date/date_linux.go:5]: I forgot to change that to date_unix.go, oos.

Comment (rminnich) [reply cmds/core/date/date_linux.go]: name changed. It builds for freebsd too, so the _linux.go made no sense.

File: cmds/core/dd/dd.go (hunk 3 above)
Comment (rminnich) [reply cmds/core/dd/dd.go]: allowedFlags

File: cmds/core/dd/dd.go (hunk 2 above)
Comment (rminnich) [reply cmds/core/dd/dd.go]: bitClearAndSet

maybe flagSetter is better but ... clear happens too ... oh well.

**Pull request merged**